RPIPs/RPIP-onchain-pdao.md

```diff
@@ -0,0 +1,457 @@
+on `RocketDAOProtocolProposals`. Where `proposalMessage` is an arbitrary string, `_payload` is an ABI encoded function
+call to `RocketDAOProtocolProposals`, `blockNumber` is the block number used to generate the merkle pollard,
+and `treeNodes` is the merkle pollard.
+
+To respond to a challenge, the proposer MUST
+call `submitRoot(uint256 proposalID, uint256 index, Types.Node[] witness, Types.Node[] nodes`
+on `RocketDAOProtocolVerifier`. Where `proposalID` is the proposal's ID, `index` is the challenged index, `witness` is
+the merkle proof and `nodes` is the new merkle pollard from the challenged index.
+
+To claim a refund for a proposal bond and rewards from any invalid challenges, the proposer MUST
+call `claimBondProposer(uint256 proposalID, uint256[] indices)` on `RocketDAOProtocolVerifier`. Where `proposalID` is
+the proposal's ID and `indices` is an array of indices that are being claimed. The index "1" refers to the proposal
+root. Claiming this index will return the proposal bond.
+
+### Verifying
+
+Any node MAY challenge a proposal by supplying an index into the merkle-sum tree that they are alleging is incorrect.
+The challenger MUST have at least `proposal.challenge.bond` unlocked effective RPL stake. And this amount of the
+challenger's RPL SHALL BE locked until the proposal is defeated.
+
+The index which is alleged to be incorrect MUST be an index of a node in one of the pollards that the proposer has
+submitted. Either the initial pollard submitted with the proposal, or any subsequent response pollards to a challenge.
+
+If the initial pollard submitted by the proposer is invalid, they will be unable to respond to challenges which require
+the proposer to provide information that is known on chain. It will take a few rounds of challenges and responses to
+reach the last layer of the merkle-sum tree. After which, the proposer will no longer be able to respond to challenges.
+
+After `proposal.challenge.period` has passed and the proposer has not responded to the challenge, any one MAY
+defeat the proposal. If a proposal is defeated, it SHALL NOT enter the voting period. And SHALL NOT be executed.
+
+After defeating a proposal, each challenger who participated in defeating the challenge SHALL be paid a proportional
+amount of the proposer's bond. Each challenger who submitted a challenge, whether it contributed to the defeat or not
+SHALL be able to claim their challenge bond.
+
+Contributing to the defeat of a proposal means a challenger submitted an index which was later proven to be incorrect by
+the proposers inability to reply to a challenge. It is possible that there are multiple incorrect indices but only those
+that resulted in the defeat of the proposal share the reward. All other challengers receive their bond back only.
+
+If a challenger challenges a node, the proposer responds, and the proposal does not get defeated. The proposer SHALL be
+able to claim the challenge bonds from the invalid challenges.
```

Comment:
This is a little interesting, but I think ok. Challenges that aren't part of the unanswered branch may or may not be right, so they aren't judged either way and simply go back 👍

Reply:
Yer, if we pay anyone who submits than there's an incentive to submit challenges that don't contribute to the actual defeat of the proposal.